main.tf

```diff
@@ -42,17 +42,12 @@ resource "aws_s3_bucket" "default" {
       }
     }
 
-    transition {
-      days          = var.standard_transition_days
-      storage_class = "STANDARD_IA"
-    }
-
     dynamic "transition" {
-      for_each = var.enable_glacier_transition ? [1] : []
+      for_each = var.enable_glacier_transition || var.enable_standard_ia_transition ? [1] : []
 
       content {
-        days          = var.glacier_transition_days
-        storage_class = "GLACIER"
+        days          = var.glacier_transition_days == null ? var.standard_transition_days : 30
```

Comment:
looks like `glacier_transition_days` is used only in the bool expression but is not actually used

Reply:
please review the logic of the two transitions, standard and glacier

Reply:
I fixed the logic, it should be good now